src/app/core/subscription-agreement.service.ts

```diff
@@ -0,0 +1,51 @@
+import { Injectable } from '@angular/core';
+
+/**
+ * @class 'SubscriptionAgreementService' is a component used to toggle the
+ * CLARK Newsletter Banner. This class is only used within @module CubeModule
+ * This class can be expanded upon; however, this class only pertains to the
+ * general user subscriptions with their email; NOT collection subscriptions.
+ */
+
+@Injectable({
+  providedIn: 'root'
+})
+export class SubscriptionAgreementService {
+  showSubscriptionBanner = true;
+
+  constructor() {}
+
+  /**
+   * Function that retrieves the toggle value for the subscription banner
+   *
+   * @returns current value of 'showSubscriptionBanner'
+   */
+  getShowSubscriptionBannerVal(): boolean {
+    this.checkShowSubscriptionBanner();
+    return this.showSubscriptionBanner;
+  }
+
+  /**
+   * @private Function to initialize local storage variable for the subscription banner
+   *
+   * @const banner checks local storage for 'showSubscriptionBanner'
+   * If null: intialize it^ @function setShowSubscriptionBanner(true)
+   */
+  private checkShowSubscriptionBanner() {
+    const banner = localStorage.getItem('showSubscriptionBanner');
+    if (banner == null) {
+      this.setShowSubscriptionBanner(true);
+    }
+    this.showSubscriptionBanner = !(banner !== null && banner === 'false');
+  }
```

Comment:
Is there a reason why you call this.setShowSubscriptionBanner() in the in but not outside? just curious.

Reply:
@JosueMolinaMorales this if statement helps to create the local storage variable if it did not already exist. The private function 'checkShowSubscriptionBanner' is hit every time we call this class so we don't want to continuously initialize the localVariable to true